src/bp/strict_encoding.rs

```diff
@@ -374,3 +374,277 @@ impl StrictDecode for bip32::ExtendedPubKey {
+        assert_eq!(written, 8);
+        assert_eq!(decoded, short_id);
+        assert!(decoded.is_offchain());
+
+        // test ShortId for OffchainTxOutput
+        let des = Descriptor::OffchainTxOutput {
+            tx_checksum: tx_check,
+            output_index: output_index,
+        };
+        let short_id = ShortId::try_from(des).unwrap();
+        let (decoded, written) = encode_decode(&short_id).unwrap();
+        assert_eq!(written, 8);
+        assert_eq!(decoded, short_id);
+        assert!(decoded.is_offchain());
+    }
+
+    #[test]
+    fn test_encoding_outpoint() {
+        let txid =
+            Txid::from_hex("217861d1a487f8e7140b9da48385e3e5d64d1ffdcd8edf0afc6818ed1331c653")
+                .unwrap();
+        let vout = 4u32;
+
+        //test random and null outpoints
+        let outpoint = OutPoint::new(txid, vout);
+        let (decoded_outpoint, written) = encode_decode(&outpoint).unwrap();
+        assert_eq!(written, 36);
+        assert_eq!(decoded_outpoint, outpoint);
+
+        let null = OutPoint::null();
+        let (decoded_null, written) = encode_decode(&null).unwrap();
+        assert_eq!(written, 36);
+        assert_eq!(decoded_null, null);
+
+        //test random and null rvealed outpoints
+        let random = OutpointReveal::from(decoded_outpoint);
+        let (decoded, written) = encode_decode(&random).unwrap();
+        assert_eq!(written, 44);
+        assert_eq!(decoded, random);
+
+        let null = OutpointReveal::from(decoded_null);
+        let (decoded, written) = encode_decode(&null).unwrap();
+        assert_eq!(written, 44);
+        assert_eq!(decoded, null);
+
+        // test random and null outpoint hash
+        let random = OutpointHash::from(decoded_outpoint);
+        let (decoded, written) = encode_decode(&random).unwrap();
+        assert_eq!(written, 32);
+        assert_eq!(decoded, random);
+
+        let null = OutpointHash::from(decoded_null);
+        let (decoded_null, written) = encode_decode(&null).unwrap();
+        assert_eq!(written, 32);
+        assert_eq!(decoded_null, null);
+    }
+
+    #[test]
+    fn test_encoding_extendedpubkey() {
+        let ext_pubkey1 = bip32::ExtendedPubKey::from_str("xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8").unwrap();
+
+        let (decoded, written) = encode_decode(&ext_pubkey1).unwrap();
+
+        assert_eq!(written, 79);
+        assert_eq!(ext_pubkey1, decoded);
+
+        let ext_pubkey2 = bip32::ExtendedPubKey::from_str("xpub6D4BDPcP2GT577Vvch3R8wDkScZWzQzMMUm3PWbmWvVJrZwQY4VUNgqFJPMM3No2dFDFGTsxxpG5uJh7n7epu4trkrX7x7DogT5Uv6fcLW5").unwrap();
+
+        let (decoded, written) = encode_decode(&ext_pubkey2).unwrap();
+
+        assert_eq!(written, 79);
+        assert_eq!(ext_pubkey2, decoded);
+    }
+
+    #[test]
+    fn test_encoding_script() {
+        let data = Vec::<u8>::from_hex(
```

Comment:
Can we pls test more scripts of different type here?

Reply:
Added.